cni/pkg/nodeagent/informers.go

```diff
@@ -218,21 +237,24 @@ func (s *InformerHandlers) reconcilePod(input any) error {
 	case controllers.EventUpdate:
 		// For update, we just need to handle opt outs
 		newPod := event.New.(*corev1.Pod)
```

Comment:
That's what we're doing in the exceptional case down below, if the pod looks like it's starting up.

I didn't do it globally because (I believe) we only need to do `s.Pods.Get` in that singular case, and not for every event.

Reply:
.Get is just a cache lookup so its not expensive - is that the concern?

Reply:
> .Get is just a cache lookup so its not expensive - is that the concern?

Yep, I don't really object to always doing it if we're comfortable doing it for every event.

Reply:
Aight, flipped it so we always just grab the most recent cache entry and categorically ignore the pod from the event.

The only way this could cause trouble is if we had 2x `podname/ns` instances with differing podUIDs, but I don't think this is actually possible.